app/assets/javascripts/host_edit.js

```diff
@@ -77,6 +75,22 @@ function computeResourceSelected(item){
   }
 }
 
+function handle_nic_updates() {
+  var modal_window = $('#interfaceModal');
+
+  var handler = function(event = null) {
+    update_nics(function() {
+      interface_subnet_selected(primary_nic_form().find('select.interface_subnet'));
+    });
+  };
+
+  if (modal_window.is(":visible")) {
+    modal_window.modal('hide').on('hidden.bs.modal', handler);
```

Comment:
It happens here: https://github.com/theforeman/foreman/blob/f391021c9c99e7b1e3949fa64c1b1fcb2c6ae540/app/assets/javascripts/host_edit.js#L22

`#interfaceModal` is a child of `#network` and the latter gets replaced completely.
This is actually the root cause of this issue and explains why we need an async approach. The modal has to be closes properly before we can remove if from the DOM.